src/main/java/upeldev/com/github/upel3/controllers/ChangePasswordController.java

```diff
@@ -0,0 +1,61 @@
+package upeldev.com.github.upel3.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import upeldev.com.github.upel3.model.User;
+import upeldev.com.github.upel3.services.UserService;
+
+import java.security.Principal;
+
+@Controller
+public class ChangePasswordController {
+
+    private final UserService userService;
+
+    @Autowired
+    public ChangePasswordController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @RequestMapping(path="/change_password")
+    public String newUser(Model model, Principal principal) {
+        User currentUser = userService.findByEmail(principal.getName());
+        model.addAttribute("user", currentUser);
+        model.addAttribute("userService", userService);
+        return "change_password";
+    }
+
+    @RequestMapping(value = "/change_password2")

```

Comment:
Why there is two "change_password"s? Wouldn't it be better to rename one from it? If in the first method the user is created the path could be smth like "new_user_password"

Reply:
Fixed.